deep_qa/data/data_generator.py

```diff
@@ -59,7 +71,18 @@ class DataGenerator:
         Only relevant if ``adaptive_batch_sizes`` is ``True``.  This is a manually-tuned parameter,
         specific to a particular model architecture and amount of GPU memory (e.g., if you change
         the number of hidden layers in your model, this number will need to change).  See
-        :func:`~TextTrainer._get_padding_memory_scaling` for more detail.
+        :func:`~TextTrainer._get_padding_memory_scaling` for more detail.  The recommended way to
+        tune this parameter is to (1) use a fixed batch size, with ``biggest_batch_first`` set to
+        ``True``, and find out the maximum batch size you can handle on your biggest instances
+        without running out of memory.  Then (2) turn on ``adaptive_batch_sizes``, and set this
+        parameter so that you get the right batch size for your biggest instances.  If you set the
+        log level to ``DEBUG`` in ``scripts/run_model.py``, you can see the batch sizes that are
+        computed.
+    maximum_batch_size: int, optional (default=1000000)
+        If we're using adaptive batch sizes, you can use this to be sure you do not create batches
+        larger than this, even if you have enough memory to handle it on your GPU.  You might
+        choose to do this to keep smaller batches because you like the noisier gradient estimates
+        that come from smaller batches, for instance.
     biggest_batch_first: bool, optional (default=False)
```

Comment:
As we start using this for different models, we should document good defaults for different GPUs in the models (just a random thought I had whilst reviewing, not really related).